### DISCUSSION TITLE
Yarn is not recognized as an internal or external command

### DISCUSSION THREAD

**Discussion by Chaw11** [warengonzaga/css-text-portrait-builder#4] [Q&A] [answered]
Hello whenever I try to do yarn install in cmd this is the message that im getting.
![cmd](https://user-images.githubusercontent.com/94739638/142721450-26983786-adad-4e0a-95de-df28e727c746.jpg)

**Comment by warengonzaga** [comment]
You need to [install NodeJS](https://nodejs.org/en/download/) first on your machine.
Then, install yarn on your machine using the command below.

```npm install --global yarn```

Check your yarn version to ensure that you have yarn on your machine.

```yarn --version```

Hope it helps!

**Reply by Chaw11** [reply]
Hi! I did those steps and when I try yarn run build this is what I get.
![sfsf](https://user-images.githubusercontent.com/94739638/142721736-d4886968-97d8-4c34-ba57-439670758fff.png)

**Reply by warengonzaga** [reply]
Download the project first here: https://github.com/WarenGonzaga/css-text-portrait-builder/archive/refs/tags/1.0.0.zip

Then unzip it, go to the folder and do yarn install and yarn run build. Before doing that make sure to edit the `config.json` file inside `src` folder inside the project folder.

Follow the complete steps here: 
https://github.com/WarenGonzaga/css-text-portrait-builder#-building-steps

**Reply by Chaw11** [reply]
I already had it downloaded and changed the config and src but I'll just redo everything to see if it'll work.

**Reply by jjaimealeman** [reply]
> `npm install --global yarn`

Aside from speed, are there any other benefits to using `yarn` instead of `npm`?

**Reply by warengonzaga** [reply]
> Aside from speed, are there any other benefits to using `yarn` instead of `npm`?

I decided to use yarn because it is faster than npm. Also considering to use pnpm.

**Comment by warengonzaga** [comment]
> I already had it downloaded and changed the config and src but I'll just redo everything to see if it'll work.

Based on the error, there is no package.json file that is supposed to be included with the package.

**Reply by Chaw11** [reply]
![image](https://user-images.githubusercontent.com/94739638/142722586-0c41b6b9-0c1b-467d-9394-90437b755e65.png)
I have the package.json file here but I still get the same error

**Reply by warengonzaga** [reply]
@Chaw11 try running ```npm --version``` and also what operating system you're using?

**Reply by Chaw11** [reply]
![image](https://user-images.githubusercontent.com/94739638/142722730-2e690f5c-dea2-4994-b947-103d4cde2d43.png)
I'm using windows 10

**Reply by warengonzaga** [reply]
Did you try using the ```npm install``` instead?
Then ```npm run build``` let me know if it will work. 
Screenshot the result.

**Reply by Chaw11** [reply]
![image](https://user-images.githubusercontent.com/94739638/142722822-c87ef901-0366-4dcc-9345-f363e8153bed.png)

**Reply by venyyyyy** [reply]
![image](https://user-images.githubusercontent.com/94595470/142722849-270edadd-01b9-4791-a832-f409451f2dac.png)
I tried npm install

**Reply by venyyyyy** [reply]
But when excuitng yarn run build/npm run build always get this
Your text portrait here, check your config file.

**Reply by warengonzaga** [reply]
I'll replicate the exact issue and get back to you guys. Since Windows 10 I can replicate it on my end.

**Reply by Chaw11** [reply]
alright. thanks!

**Reply by warengonzaga** [reply]
I just found out that before you ``yarn install`` and ``yarn run build`` be sure that you are in the directory of the project.

![image](https://user-images.githubusercontent.com/15052701/142723109-44ec8514-4138-4ea5-9eef-7ec19bedafb5.png)

**Reply by warengonzaga** [reply]
One note: before doing the ``yarn run build`` be sure to delete the ``dist`` folder or the old and the new build will mix. I'll this update real quick.

**Reply by warengonzaga** [reply]
@Chaw11 try running ``npm run dev`` since the local build will not work since it is not intended to view in a local setup. I'll work on packaging so everyone can use it even without being served.

**Reply by Chaw11** [reply]
I'm not getting any error now, I just get the same as @venyyyyy after doing yarn run build when I open index

**Reply by warengonzaga** [reply]
Serve the build by doing ``yarn run dev or npm run dev`` it should work fine now. The local build is not available for now I need to confirm this with the team behind Yarn package manager.

**Reply by Chaw11** [reply]
oh alright, got it. It worked now

**Reply by warengonzaga** [reply]
That's good news! @Chaw11 let me work on the next update to support local builds.

**Comment by venyyyyy** [comment]
> Hi! I did those steps and when I try yarn run build this is what I get. ![sfsf](https://user-images.githubusercontent.com/94739638/142721736-d4886968-97d8-4c34-ba57-439670758fff.png)

try to install node 12.18.3 I got a error at first because I downloaded 12.13.0

**Reply by Chaw11** [reply]
did it work for you alr?

**Comment by venyyyyy** [comment]
> I just found out that before you `yarn install` and `yarn run build` be sure that you are in the directory of the project.
> 
> ![image](https://user-images.githubusercontent.com/15052701/142723109-44ec8514-4138-4ea5-9eef-7ec19bedafb5.png)

yup i already did that always getting this  Your text portrait here, check your config file.

**Reply by warengonzaga** [reply]
Currently, working on a fix. This is confirmed as a bug on my end.

**Reply by venyyyyy** [reply]
> I just found out that before you `yarn install` and `yarn run build` be sure that you are in the directory of the project.
> 
> ![image](https://user-images.githubusercontent.com/15052701/142723109-44ec8514-4138-4ea5-9eef-7ec19bedafb5.png)

thanks

**Reply by warengonzaga** [reply]
Hey, @venyyyyy as per checking the local build will not work. Unless you serve it, so you need to use ``yarn run dev`` to run the project.

**Comment by warengonzaga** [comment, marked as answer]
Hi guys, working on a hotfix right now. A new version will be released in a bit. Please bear with me. 😇😅

Make sure you have `Yarn` on your machine.

```bash
npm install --global yarn
```

and then type:

```bash
yarn --version
```

This will make sure that you have Yarn installed on your machine.

**Update**: I revise the build step in version 1.0.1, just follow the command in order.

- ``yarn install``
- ``yarn build``
- ``yarn serve``

Note: This will change in the coming updates, always double-check what version you're using.

**Reply by Jepoy1221** [reply]
@parcel/transformer-json: JSON5: invalid character '\n' at 4:0 how to fix this sir

**Reply by warengonzaga** [reply]
@Jepoy1221 your JSON file is invalid use this to validate your JSON file. https://jsonlint.com/
Copy-paste your entire JSON file content and paste it there.
